test/medley/core_test.cljc

```diff
@@ -355,6 +355,26 @@
                       [1 3 5 6 8 9])
            [[1 3 5] [6]]))))
 
+(deftest test-partition-with
+  (testing "sequences"
+    (is (= (m/partition-with = nil) '()))
+    (is (= (m/partition-with < [1 1 2 2 1 1 3 3 3]) '((1 1) (2 2 1 1) (3 3 3))))
+    (is (= (m/partition-with < [1 2 3 4 5]) '((1) (2) (3) (4) (5))))
+    (is (= (m/partition-with < [5 4 3 2 1]) '((5 4 3 2 1)))))
+
+  (testing "transducers"
+    (is (= (transduce (m/partition-with <) conj nil) []))
```

Comment:
As above, let's also check the case where there is one element in the collection.